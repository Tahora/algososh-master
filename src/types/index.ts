import {FC, PropsWithChildren} from "react";

export type FCC<P = {}> = FC<PropsWithChildren<P>>;

export enum SortVisualizerLength {
    Short = "shortLength",
    Medium = "mediumLength",
    Large = "largeLength",
}

export enum SortAlgorithms {
    Bubble = "bubble",
    Selection = "selection",
}

export enum LoaderTypes {
    PushHead = "pushHead",
    PushTail = "pushTail",
    RemoveHead = "removeHead",
    RemoveTail = "removeTail",
    InsertByIndex = "insertByIndex",
    RemoveByIndex = "RemoveByIndex"
}

export enum StackLoaderTypes {
    Push = "push",
    Remove = "remove",
    Clean = "clean",
}


