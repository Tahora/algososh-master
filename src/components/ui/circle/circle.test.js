import React from 'react';
import renderer from 'react-test-renderer';
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";


describe('Circle test', function () {
        it('without letter', () => {
            const tree = renderer
                .create(<Circle/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with letter', () => {
            const tree = renderer
                .create(<Circle letter="l"/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with head', () => {
            const tree = renderer
                .create(<Circle head="head"/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with react component in head', () => {
            const tree = renderer
                .create(<Circle head={<Circle/>}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with tail', () => {
            const tree = renderer
                .create(<Circle tail="tail"/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with react component in tail', () => {
            const tree = renderer
                .create(<Circle tail={<Circle/>}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('with index', () => {
            const tree = renderer
                .create(<Circle index={0}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('small', () => {
            const tree = renderer
                .create(<Circle isSmall={true}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('state default', () => {
            const tree = renderer
                .create(<Circle state={ElementStates.Default}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('state changing', () => {
            const tree = renderer
                .create(<Circle state={ElementStates.Changing}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('state modified', () => {
            const tree = renderer
                .create(<Circle state={ElementStates.Modified}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

    }
)
