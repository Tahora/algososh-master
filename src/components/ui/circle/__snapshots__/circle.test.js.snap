// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle test small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test state changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test state default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test state modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle test with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      l
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test with react component in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test with react component in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle test with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle test without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
